src/Plugin/BulletinBase.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Drupal\govdelivery_bulletins\Plugin;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+
+/**
+ * Provides base functionality for the NodePublish Queue Workers.
+ */
+class BulletinBase extends QueueWorkerBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * BulletinBase constructor.
+   */
+  public function __construct() {
+
+  }
+
+  /**
+   * Processes the bulletins in the queue.
+   *
+   * @param DateTime $time
+   *   The timestamp (optional) of when the queue item is to be actionalble.
+   */
+  public function processQueue($time = NULL) {
+    // Call our queue service.
+    $queue_factory = \Drupal::service('queue');
+    $queue_manager = \Drupal::service('plugin.manager.queue_worker');
+
+    // Call bulletins service, and create an instance for processing.
+    $queue = $queue_factory->get('govdelivery_bulletins');
+    $queue_worker = $queue_manager->createInstance('govdelivery_bulletins');
+
+    // Get the number of items.
+    $number_of_queue = $queue->numberOfItems();
+
+    for ($i = 0; $i < $number_of_queue; $i++) {
+      // Get a queued item.
+      $item = $queue->claimItem();
+      if ($time && $item->created < $time || empty($time)) {
+        // Process it.
+        $queue_worker->processItem($item->data);
+        // Call the send service.
+        // If everything was correct, delete the processed item from the queue.
+        $queue->deleteItem($item);
+      }
+      elseif ($time && $item->created > $time) {
```

Comment:
```suggestion
      else {
```
I think we want this always to be the fallback if the first condition fails.

Reply:
:+1: